modules/python/package/cv2/__init__.py

```diff
@@ -15,16 +16,56 @@
     print('    pip install numpy')
     raise
 
+# TODO
+# is_x64 = sys.maxsize > 2**32
+
 
-py_code_loader = None
-if sys.version_info[:2] >= (3, 0):
+def __load_extra_py_code_for_module(base, name, enable_debug_print=False):
+    module_name = "{}.{}".format(__name__, name)
+    export_module_name = "{}.{}".format(base, name)
+    native_module = sys.modules.pop(module_name, None)
     try:
-        from . import _extra_py_code as py_code_loader
-    except:
-        pass
+        m = importlib.import_module(module_name)
+    except ImportError as err:
+        if enable_debug_print:
+            print("Can't load Python code for module:", module_name,
+                  ". Reason:", err)
+        # Extension doesn't contain extra py code
+        return False
+
+    if hasattr(m, '__all__'):
+        export_members = {k: getattr(m, k) for k in m.__all__}
+    else:
+        export_members = m.__dict__
+
+    if not hasattr(base, name):
+        setattr(sys.modules[base], name, m)
+    sys.modules[export_module_name] = m
+    # If it is C extension module it is already loaded by cv2 package
+    if native_module:
+        for k, v in native_module.__dict__.items():
+            if enable_debug_print: print('    symbol: {} = {}'.format(k, v))
+            setattr(sys.modules[export_module_name], k, v)
```

Comment:
I believe Python extension code should have priority over C binary extension implementation.
Currently it is unconditionally overwritten from "native" module.

Reply:
Regardless the strategy we choose other function implementation wan't be available.
I would prefer to raise exception in this case or at least add a [warning message](https://docs.python.org/3/library/warnings.html#warnings.warn), because of this. What do you think?

Reply:
1. Bindings are usually declared for multiple languages at once.
2. Python extra code may improve that.
3. We should be able to replace functionality provided by C binary extension.
4. Python extra code should have access for underlying C binary implementation (to properly wrap that), perhaps we should store "native" module somewhere as we replacing it with the package (e.g., `cv2._native`).